packages/components/src/slot-fill/bubbles-virtually/slot-fill-provider.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useCallback, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import SlotFillContext from './slot-fill-context';
+
+function useSlotRegistry() {
+	const [ slots, setSlots ] = useState( {} );
+	const [ fills, setFills ] = useState( {} );
+
+	const registerSlot = useCallback( ( name, ref, fillProps ) => {
+		setSlots( ( prevSlots ) => ( {
+			...prevSlots,
+			[ name ]: {
+				...prevSlots[ name ],
+				ref: ref || prevSlots[ name ].ref,
+				fillProps: fillProps || prevSlots[ name ].fillProps || {},
+			},
+		} ) );
+	}, [] );
+
+	const unregisterSlot = useCallback( ( name, ref ) => {
+		setSlots( ( prevSlots ) => {
+			// eslint-disable-next-line no-unused-vars
+			const { [ name ]: slot, ...nextSlots } = prevSlots;
+			// Make sure we're not unregistering a slot registered by another element
+			// See https://github.com/WordPress/gutenberg/pull/19242#issuecomment-590295412
+			if ( slot.ref === ref ) {
```

Comment:
Should it be safe to assume `slot` is defined here?

In debugging the issue mentioned by @draganescu at https://github.com/WordPress/gutenberg/issues/18560#issuecomment-604920204 , the exact error occurs at this line of code:

![image](https://user-images.githubusercontent.com/1779930/77765354-874f4f00-7014-11ea-97a2-54cf7c32cc24.png)

It's not yet clear to me if the bug is in how the slot came to be `undefined` in the first place, or if we should be guarded in how we expect `slot` to be available at this line of code.